src/components/ImageView/index.js

```diff
@@ -1,35 +1,117 @@
-import React, {memo} from 'react';
+import React, {PureComponent} from 'react';
 import PropTypes from 'prop-types';
-import {View, Image} from 'react-native';
-import styles from '../../styles/styles';
+import {View, Image, Pressable} from 'react-native';
+import styles, {getZoomCursorStyle, getZoomSizingStyle} from '../../styles/styles';
 
 const propTypes = {
     // URL to full-sized image
     url: PropTypes.string.isRequired,
 };
 
-const ImageView = props => (
-    <View
-        style={[
-            styles.w100,
-            styles.h100,
-            styles.alignItemsCenter,
-            styles.justifyContentCenter,
-            styles.overflowHidden,
-        ]}
-    >
-        <Image
-            source={{uri: props.url}}
-            style={[
-                styles.w100,
-                styles.h100,
-            ]}
-            resizeMode="center"
-        />
-    </View>
-);
+class ImageView extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.scrollableRef = null;
+        this.state = {
+            isZoomed: false,
+            isDragging: false,
+            isMouseDown: false,
+            initialScrollLeft: 0,
+            initialScrollTop: 0,
+            initialX: 0,
+            initialY: 0,
+        };
+    }
 
-ImageView.propTypes = propTypes;
-ImageView.displayName = 'ImageView';
+    componentDidMount() {
+        document.addEventListener('mousemove', this.trackMouseMove.bind(this));
+        document.addEventListener('touchmove', this.trackMouseMove.bind(this));
```

Comment:
Why did you need to set up a listener for `touchmove`?

Reply:
So that mobile web touches would set `isDragging`

Reply:
But maybe there is a solution where we don't need to do this and we can tell what kind of mouse event we've got.